pkg/controller/kibana/stackmon/sidecar.go

```diff
@@ -60,20 +61,38 @@ func Metricbeat(ctx context.Context, client k8s.Client, kb kbv1.Kibana, basePath
 		}
 	}
 
-	metricbeat, err := stackmon.NewMetricBeatSidecar(
-		ctx,
-		client,
-		commonv1.KbMonitoringAssociationType,
-		&kb,
-		kb.Spec.Version,
-		metricbeatConfigTemplate,
-		kbv1.KBNamer,
-		fmt.Sprintf("%s://localhost:%d", kb.Spec.HTTP.Protocol(), network.HTTPPort),
-		basePath,
-		username,
-		password,
-		kb.Spec.HTTP.TLS.Enabled(),
-	)
+	v, err := version.Parse(kb.Spec.Version)
+	if err != nil {
+		return stackmon.BeatSidecar{}, err // error unlikely and should have been caught during validation
+	}
+	caVol, err := stackmon.CAVolume(client, k8s.ExtractNamespacedName(&kb), kbv1.KBNamer, commonv1.KbMonitoringAssociationType, kb.Spec.HTTP.TLS.Enabled())
+	if err != nil {
+		return stackmon.BeatSidecar{}, err
+	}
+	type inputConfigData struct {
```

Comment:
cosmetic nit, we are also sometimes using an anonymous struct, I was wondering if we should make this type a first class citizen in `stackmon`.